pipeline/file_pulling.go

```diff
@@ -0,0 +1,152 @@
+package pipeline
```

Comment:
`pipeline` is a bit different concept.  It's just a generic container that can have whatever elements / engines as part of it.  F.e. it can have a DeviceDetectionCloudRequest engine and a JsonElement and JavascriptBuilderElement.  Or it can have a DeviceDetectionOnPremise engine and some other element that pre-processes evidence...  What you've built here is essentially OnPremise device detection functionality.  It roughly corresponds to DeviceDetectionOnPremise class in node.js implementation:  https://github.com/51Degrees/device-detection-node/tree/main/fiftyone.devicedetection.onpremise. Let's thus rename this package into `onpremise`.  This API can live under `dd` namespace - so full path would be `dd.onpremise` package.  Later there might be needed to add `dd.cloud` package - for the cloud device-detection.  

The main class name should be not `Pipeline`, but rather `DeviceDetectionOnPremise` or `DeviceDetectionOnPremiseEngine`, later we'll add `DeviceDetectionCloudEngine`.

Reply:
since package is onpremise, and whole library is called device detection, i think it should just be called Engine

since when imported it says onpremise.Engine.
even my IDE gives warning: ``Name starts with the package name ``

Reply:
sounds good

Reply:
```
package main

import (
	"github.com/51Degrees/device-detection-go/v4/dd"
	"github.com/51Degrees/device-detection-go/v4/onpremise"
	"log"
)

func main() {
	manager := dd.NewResourceManager()
	config := dd.NewConfigHash(dd.Balanced)

	pl, err := onpremise.New(
		manager,
		config,
		onpremise.WithDataUpdateUrl(
			"https://myprovider.com/1.tar.gz",
			30000,
		),
	)

	if err != nil {
		log.Fatalf("Failed to create pipeline: %v", err)
	}

	resultsHash := pl.NewResultsHash(0, 1)
	//use results and do detection
	resultsHash.Free()
	<-make(chan struct{})

	defer manager.Free()
}
```